lib/pages/home/home_page.dart

```diff
@@ -22,21 +27,137 @@ class HomePage extends StatelessWidget {
           route: Routes.settings,
         )
       ],
-      body: Center(
-        child: FutureBuilder(
-          future: authProvider.currentUser,
-          builder: (BuildContext context, AsyncSnapshot<User> snap) {
-            String firstName;
-            if (snap.hasData) {
-              User user = snap.data;
-              firstName = user.firstName;
-            }
-            return Text(firstName == null
-                ? S.of(context).messageWelcomeSimple
-                : S.of(context).messageWelcomeName(firstName));
-          },
-        ),
+      body: ListView(
+        children: [
+          favouriteWebsites(context),
+        ],
       ),
     );
   }
+
+  Padding favouriteWebsites(BuildContext context) {
+    var websites = Provider.of<WebsiteProvider>(context, listen: false)
+        .fetchWebsites(null);
+
+    return Padding(
+        padding: const EdgeInsets.only(left: 8.0, right: 8.0, top: 8.0),
+        child: FutureBuilder(
+            future: websites,
+            builder: (_, snapshot) {
+              if (snapshot.connectionState == ConnectionState.done) {
+                List<Website> websitesData = snapshot.data;
```

Comment:
Filter the data here, not when building the widget.

```
websites = snapshot.data.where((w) => w.numberOfVisits > 0).take(3);
```

And change the if below to `if (websites.isEmpty) noneYet() else ...`